tidy3d/web/api/mode.py

```diff
@@ -83,6 +85,21 @@ def run(
     if verbose:
         console = get_logging_console()
 
+    if reduce_simulation == "auto":
+        sim = mode_solver.simulation
+        contains_custom = any(isinstance(s.medium, AbstractCustomMedium) for s in sim.structures)
+        contains_custom = contains_custom or isinstance(sim.medium, AbstractCustomMedium)
+        reduce_simulation = contains_custom
+
+        if reduce_simulation:
+            log.warning(
+                "The associated `Simulation` object contains custom mediums. It will be "
+                "automatically restricted to the mode solver plane to reduce data for uploading. "
+                "To force uploading the original `Simulation` object use `reduce_simulation=False`."
+                " Setting `reduce_simulation=True` will force simulation reduction in all cases and"
+                " silence this warning."
+            )
+
```

Comment:
I think it looks good, probably we should update the notebooks then to avoid the warning?

Reply:
was there any other besides BullseyeCavityPSO?

Reply:
BullseyeCavityPSO will not have a warning anymore as it doesn't use custom medium, I was thinking of the heat notebooks actually.